pandas/tests/frame/test_to_csv.py

```diff
@@ -1220,6 +1220,18 @@ def test_multi_index_header(self):
         expected = tm.convert_rows_list_to_csv_str(expected_rows)
         assert result == expected
 
+    def test_to_csv_single_level_multi_index(self):
+        # see gh-26303
+        index = pd.Index([(1,), (2,), (3,)])
+        df = pd.DataFrame([[1, 2, 3]], columns=index)
+        with ensure_clean() as path:
+            df = df.reindex(columns=[(1,), (3,)])
+            df.to_csv(path, line_terminator='\n')
+            expected = b",1,3\n0,1,3\n"
+
+            with open(path, mode='rb') as f:
```

Comment:
Do you need to physically write this to a file to reproduce the segfault or can you just inspect the result returned as a string?

Reply:
Tried the example provided in the issue. The segfault occured even if the csv was never written to a file and the test now just writes to a string.